client/oslc-client/src/main/java/org/eclipse/lyo/client/OslcClient.java

```diff
@@ -33,577 +29,477 @@
-												return creationFactory;
-											}
-										}
-									} else {
-										return creationFactory;
-									}
-								}
-							}
-							//Check if this is the default factory
-							for (URI usage:creationFactory.getUsages()) {
-								if (usage.toString() != null && usage.toString().equals(OSLCConstants.USAGE_DEFAULT_URI)) {
-									defaultCreationFactory = creationFactory;
-								}
-							}
-						}
-					}
-				}
-			}
-		}
-
-		//If we reached this point, there was no resource type match
-		if (defaultCreationFactory != null) {
-			//return default, if present
-			return defaultCreationFactory;
-		} else if (firstCreationFactory != null && firstCreationFactory.getResourceTypes().length ==0) {
-			//return the first for the domain, if present
-			return firstCreationFactory;
-		}
-
-		throw new ResourceNotFoundException(serviceProviderUrl, "CreationFactory");
-	}
-
-	/**
-	 * Find the OSLC Creation Factory URL for a given OSLC resource type.  If no resource type is given, returns
-	 * the default Creation Factory, if it exists.
-	 *
-	 * @param oslcResourceType - the resource type of the desired query capability.   This may differ from the OSLC artifact type.
-	 * @return URL of requested Creation Factory or null if not found.
-	 */
-	public String lookupCreationFactory(final String serviceProviderUrl, final String oslcDomain, final String oslcResourceType)
-			throws IOException, URISyntaxException, ResourceNotFoundException
-	{
-		return lookupCreationFactory(serviceProviderUrl, oslcDomain, oslcResourceType, null);
-	}
-
-	/**
-	 * Find the OSLC Creation Factory URL for a given OSLC resource type and OSLC usage.  If no resource type is given, returns
-	 * the default Creation Factory, if it exists.
-	 *
-	 * @param oslcResourceType - the resource type of the desired query capability.   This may differ from the OSLC artifact type.
-	 * @return URL of requested Creation Factory or null if not found.
-	 */
-	public String lookupCreationFactory(final String serviceProviderUrl, final String oslcDomain, final String oslcResourceType, final String oslcUsage)
-			throws IOException, URISyntaxException, ResourceNotFoundException
-	{
-		return lookupCreationFactoryResource(serviceProviderUrl, oslcDomain, oslcResourceType, oslcUsage).getCreation().toString();
-	}
-
-
-}
+        return getResource(url, requestHeaders, defaultMediaType, configurationContext, true);
+    }
+
+    public Response getResource(String url, Map<String, String> requestHeaders, String defaultMediaType,
+            boolean handleRedirects) {
+        return getResource(url, requestHeaders, defaultMediaType, null, handleRedirects);
+    }
+
+    public Response getResource(String url, Map<String, String> requestHeaders, String defaultMediaType,
+            String configurationContext, boolean handleRedirects) {
+        return doRequest(HttpMethod.GET, url, null, configurationContext, null, defaultMediaType, OSLCConstants.CT_RDF,
```

Comment:
This is a significant refactoring, and looks promising. But I can't easily review it. Too many other (formatting) changes in the way.
If you are confortable with that @berezovskyi, I am fine.